app/services/geek_jokes_api_service.js

```diff
@@ -0,0 +1,11 @@
+const axios = require('axios');
+
+const base_api_url = `${process.env.BASE_API_URL_GEEK_JOKES}/api?format=json`;
+
+exports.geekJokesApi = {
+  response: () =>
+    axios
```

Comment:
Using axios it's very well good job. Personally I like to create a helper folder. In the helper folder I usually create a file or folder name htt_clint then I create a generic function an this use a axios. I think that is a good idea because you should changed axios for some reason you just have to change it in this place. For example 

**_app/helpers/http-client/index.js_**


```
const request = require('axios');
const { httpRequestClientErrorHandler } = require('./errors');

exports.buildConfig = ({ method = 'get', host, endpoint, data, headers, params }) => ({
  method,
  url: `${host}/${endpoint}`,
  headers: {
    'content-type': 'application/json',
    ...headers
  },
  data,
  params
});

exports.request = options =>
  request(options)
    .then(res => res.data)
    .catch(error => httpRequestClientErrorHandler(error));
```

**_app/services/auth0.js_**

```
const requestAuth0 = options => {
  const buildOptions = buildConfig({
    ...options,
    host: auth0Config.auth0endpoint
  });
  return request(buildOptions);
};

exports.getAccessToken = () => {
  const options = {
    method: 'post',
    endpoint: 'oauth/token',
    data: {
      grant_type: 'client_credentials',
      client_id: auth0Config.clientId,
      client_secret: auth0Config.clientSecret,
      audience: `${auth0Config.auth0endpoint}/api/v2/`
    }
  };
  return requestAuth0(options);
};
```